tests/unit/modules/network/sonic/fixtures/sonic_lldp_interfaces.yaml

```
@@ -27,23 +34,45 @@ merged_01:
                   - openconfig-lldp-ext:MED_NETWORK_POLICY
                   - openconfig-lldp-ext:MED_EXT_MDI_POWER
                   - openconfig-lldp-ext:MDI_POWER
-                  
+                  - openconfig-lldp-ext:PORT_VLAN_ID
+                  - openconfig-lldp-ext:VLAN_NAME
+                  - openconfig-lldp-ext:LINK_AGGREGATION
+                  - openconfig-lldp-ext:MAX_FRAME_SIZE
+                  - openconfig-lldp-ext:VLAN_NAME
+
   expected_config_requests:
     - path: data/openconfig-lldp:lldp/interfaces/interface=Ethernet0/config/openconfig-lldp-ext:mode
       method: patch
-      data: 
+      data:
         openconfig-lldp-ext:mode: 'RECEIVE'
     - path: data/openconfig-lldp:lldp/interfaces/interface=Ethernet0/config/openconfig-lldp-ext:management-address-ipv4
       method: patch
-      data: 
+      data:
         openconfig-lldp-ext:management-address-ipv4: '10.1.1.1'
     - path: data/openconfig-lldp:lldp/interfaces/interface=Ethernet0/config/openconfig-lldp-ext:suppress-tlv-advertisement=MDI_POWER
       method: delete
     - path: data/openconfig-lldp:lldp/interfaces/interface=Ethernet0/config/openconfig-lldp-ext:suppress-tlv-advertisement=MED_EXT_MDI_POWER
       method: delete
     - path: data/openconfig-lldp:lldp/interfaces/interface=Ethernet0/config/openconfig-lldp-ext:suppress-tlv-advertisement=MED_NETWORK_POLICY
       method: delete
-      
+    - path: data/openconfig-lldp:lldp/interfaces/interface=Ethernet0/config/openconfig-lldp-ext:suppress-tlv-advertisement=PORT_VLAN_ID
+      method: delete
+    - path: data/openconfig-lldp:lldp/interfaces/interface=Ethernet0/config/openconfig-lldp-ext:suppress-tlv-advertisement=VLAN_NAME
+      method: delete
+    - path: data/openconfig-lldp:lldp/interfaces/interface=Ethernet0/config/openconfig-lldp-ext:suppress-tlv-advertisement=LINK_AGGREGATION
+      method: delete
+    - path: data/openconfig-lldp:lldp/interfaces/interface=Ethernet0/config/openconfig-lldp-ext:suppress-tlv-advertisement=MAX_FRAME_SIZE
+      method: delete
+    - path: data/openconfig-lldp:lldp/interfaces/interface=Ethernet0/config/openconfig-lldp-ext:allowed-vlans
+      method: patch
+      data:
+        openconfig-lldp-ext:allowed-vlans: '10,15-20'
```

This format specification is incorrect. It looks like the functional code is correct and works, but the format shown here isn't the format in which the code in the "config" file would send this request. If it did, the request would be rejected. The value associated with the allowed_vlans key should be a list containing unquoted int members and quoted ranges delimited by ".." (not "-"). This line should be:

```suggestion
        openconfig-lldp-ext:allowed-vlans:
            - 10
            - '15..20'
```
This means that the unit test should be failing here. If it isn't failing on this, the reason for not detecting the failure needs to be fixed.

---

addressed

---

The current code implementation and UT expectations match as required. The previously missing validation has been fixed.

Resolved.